From 0463a9a72a906b7feb7c71cea10634e3dc466dd1 Mon Sep 17 00:00:00 2001
From: "Bandi, Kushal" <kushal.bandi@intel.com>
Date: Mon, 2 Jul 2018 15:47:41 -0700
Subject: [PATCH 08/16] media: intel-ipu4: [VIRT] Fixing buffer lookup issue

With every GET_BUF new instance of buffer was getting created.
Because the look-up was not looking in to appropriate list.

Change-Id: Ie580f5358d377b698edcc8c20026fe403cac002c
Signed-off-by: Bandi, Kushal <kushal.bandi@intel.com>
---
 drivers/media/pci/intel/virtio/intel-ipu4-para-virt-drv.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/media/pci/intel/virtio/intel-ipu4-para-virt-drv.c b/drivers/media/pci/intel/virtio/intel-ipu4-para-virt-drv.c
index 8a79bce..12d6f34 100644
--- a/drivers/media/pci/intel/virtio/intel-ipu4-para-virt-drv.c
+++ b/drivers/media/pci/intel/virtio/intel-ipu4-para-virt-drv.c
@@ -110,7 +110,7 @@ static struct ici_frame_buf_wrapper *frame_buf_lookup(struct ici_isys_frame_buf_
 	int i;
 	int mem_type = user_frame_info->mem_type;
 
-	list_for_each_entry(buf, &buf_list->getbuf_list, node) {
+	list_for_each_entry(buf, &buf_list->getbuf_list, uos_node) {
 		for (i = 0; i < user_frame_info->num_planes; i++) {
 			struct ici_frame_plane *new_plane =	&user_frame_info->frame_planes[i];
 			struct ici_frame_plane *cur_plane =	&buf->frame_info.frame_planes[i];
@@ -293,6 +293,7 @@ struct ici_frame_buf_wrapper *get_buf(struct virtual_stream *vstream, struct ici
 		buf->state = ICI_BUF_PREPARED;
 		return buf;
 	}
+	pr_debug("Creating new buffer in the list\n");
 	buf = kzalloc(sizeof(*buf), GFP_KERNEL);
 	if (!buf)
 		return NULL;
-- 
1.9.1

